core/patina_internal_collections/src/node.rs

```diff
@@ -275,86 +266,86 @@ where
     D: SliceKey,
 {
     fn set_color(&self, color: bool) {
-        self.color.store(color, Ordering::SeqCst);
+        self.color.set(color);
     }
 
     fn is_red(&self) -> bool {
-        self.color.load(Ordering::SeqCst) == RED
+        self.color.get() == RED
     }
 
     fn is_black(&self) -> bool {
-        self.color.load(Ordering::SeqCst) == BLACK
+        self.color.get() == BLACK
     }
 
     fn color(&self) -> bool {
-        self.color.load(Ordering::SeqCst)
+        self.color.get()
     }
 
     fn parent(&self) -> Option<&Node<D>> {
-        let node = self.parent.load(Ordering::SeqCst);
+        let node = self.parent.get();
         match node.is_null() {
             true => None,
-            false => Some(unsafe { &*node }),
+            false => Some(unsafe { node.as_mut().expect("node is not null") }),
```

Comment:
`&Node<D>` is a shared reference. `node.as_mut()` is returning a mutable reference. Is that necessary?

Could something like `node.as_ref()` work?

Reply:
Good point; I'll switch it to as_ref(). I think due to the signature of the method, the caller still receives an immutable ref.